modules/core/src/main/scala/gem/Location.scala

```diff
@@ -23,6 +23,14 @@ sealed trait Location extends Product with Serializable {
     * if any.  Beginning and End Locations have no fixed elements.
     */
   protected def minPrefixLength: Int
+
+  final override def toString: String =
+    this match {
+      case Location.Beginning  => "{α}"
+      case Location.Middle(ps) => "{" + ps.map(_.toString).intercalate(",") + "}"
+      case Location.End        => "{ω}"
+    }
```

Comment:
This fixes the `ToString` wart because now `Location` implements `.toString` directly. In principle it should be smart enough to say well it's sealed and every subclass implements `toString` but It can't do that. This seems fine to me though.